src/common/document/parser.ts

```diff
@@ -18,6 +24,14 @@ export function isValidDocumentIdentifier(str: string): boolean {
   return ID_NAME_RE.test(str);
 }
 
+const PROVIDER_ID_RE = /^(unverified!)?[a-z][a-z_-]*$/;
```

Comment:
My question here is whether we want to parse prefixes _in general_ and then check for validity of the `unverified` prefix or whether we are directly encoding the idea of only one existing prefix into the parser.

In the first case I'd allow any valid lowercase id as the prefix and then check later.
In the second case I'd change the error message "${provider} is not a valid lowercase identifier" because to me it now sounds wrong.

Reply:
It probably will be different. So lets wait a bit.